### ISSUE TITLE
nest import causes scikit-learn to segfault

### ISSUE THREAD

**Issue by maharjun** [nest/nest-simulator#848]
The following is the code that can reproduce the error

```python
import nest
from sklearn.svm import LinearSVC

import numpy as np

X_data = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
Y_data = np.array([0, 0, 1])
linsvc = LinearSVC(C=1.0, class_weight={0: 0.001, 1: 2.0}).fit(X=X_data, y=Y_data)
```

This code will segfault. However, if you flip the order of the imports to
```python
from sklearn.svm import LinearSVC
import nest
```

Then everything works just fine.

I have dug into the issue a bit and uncovered this.

scikit-learn imports cython-compiled libraries libsvm and liblinear both of which contain a (non-exposed) function named `set_parameter` which is called when one makes an invocation to `SVC.fit()` or `LinearSVC.fit()` respectively. The problem is that if nest is imported before scikit-learn, then it causes scikit-learn to mess up the dynamic loading somehow (This is the part that I don't know) which leads to the `liblinear.train_wrap` function that is called by `LinearSVC.fit()` to use the `set_parameter` function from libsvm instead of liblinear. The parameters of the two varieties of `set_parameter` are different thus causing a segfault. It appears that this does not happen if nest is imported after sklearn.svm.

This can be seen by setting a breakpoint on the function `set_parameter` when running the python script under GDB.

This is the first case (i.e. nest before sklearn.svm => segfault). The following is the output of gdb when it reaches a function named `set_parameter`
```
Breakpoint 1, set_parameter (param=0x1, svm_type=2, kernel_type=24510720, degree=26374064, gamma=0.0001, coef0=1, nu=0.10000000000000001, cache_size=0, 
    C=1.3852388523421298e-309, eps=0, p=0, shrinking=1000, probability=686439501, nr_weight=-11664, weight_label=0x0, weight=0x0, max_iter=686439501, 
    random_seed=-563176512) at sklearn/svm/src/libsvm/libsvm_helper.c:74
```

This is the second case (i.e. nest after sklearn.svm => no segfault). The following is the output of gdb when it reaches a function named `set_parameter`

```
Breakpoint 1, set_parameter (solver_type=1, eps=0.0001, C=1, nr_weight=2, weight_label=0x10aad20 "", weight=0x160e8d0 "\374\251\361\322MbP?", 
    max_iter=1000, seed=277763991, epsilon=0.10000000000000001) at sklearn/svm/src/liblinear/liblinear_helper.c:190
```

We can see in the second case that the correct function is called where-as in the first case, an incorrect function is called.

Why would a nest import cause such a thing?

**Comment by terhorstd** [comment]
We have seen a similar behaviour with importing `scipy`. The current workaround is to import `scipy` first, as you do with `sklearn`.

**Comment by heplesser** [comment]
Fixed by #891.
